index.d.ts

```diff
@@ -37,6 +44,27 @@ await pWaitFor(() => pathExists('unicorn.png'));
 console.log('Yay! The file now exists.');
 ```
 */
-export default function pWaitFor(condition: () => PromiseLike<boolean> | boolean, options?: Options): Promise<void>;
+export default function pWaitFor<ResolveValueType>(condition: () => PromiseLike<boolean> | boolean | ResolveValue<ResolveValueType> | PromiseLike<ResolveValue<ResolveValueType>>, options?: Options): Promise<ResolveValueType>;
+
+/**
+Resolve the main promise with a custom value.
+
+@example
+```
+import pWaitFor, {resolveWith} from 'p-wait-for';
+import pathExists from 'path-exists';
+
+const path = await pWaitFor(async () => {
+	const path = getPath();
+
+	if (await pathExists(path)) {
+		return resolveWith(path);
+	}
```

Comment:
I think there needs to be an explicit `return false` in this code example to avoid a `TypeError('Expected condition to return a boolean');`

Reply:
No, `resolveWith` returns a custom object to resolve with. This object is checked for before any boolean assertion is run.

Reply:
Oh, I meant a `return false` for when the if-statement is false

Reply:
I see, good spotting!